core/include/kernel/interrupt.h

```diff
@@ -287,11 +319,137 @@ TEE_Result interrupt_alloc_add_handler(struct itr_chip *chip, size_t it_num,
+ */
+typedef struct itr_line *(*dt_get_itr_func)(struct dt_pargs *args, void *data,
+					    TEE_Result *res);
+
+#ifdef CFG_DT
+/**
+ * dt_register_interrupt_provider - Register an interrupt provider
+ *
+ * @fdt		Device tree to work on
+ * @node	Node offset of the interrupt controller in the DT
+ * @dt_get_itr	Callback to match the devicetree interrupt reference with
+ * @data	Data which will be passed to the get_dt_its callback
+ */
+TEE_Result dt_register_interrupt_provider(const void *fdt, int node,
+					  dt_get_itr_func dt_get_itr,
+					  void *data);
+
+/**
+ * dt_get_interrupt_by_index() - Get an interrupt from DT by interrupt index
+ *
+ * @fdt		Device tree to work on
+ * @node	Node offset of the subnode containing interrupt(s) references
+ * @index	Index in "interrupts" or "exteneded-interrupts" property list
+ * @chip	Output interrupt controller clock reference upon success
+ * @itr_num	Output interrupt line upon success
+ *
+ * Return TEE_SUCCESS in case of success
+ * Return TEE_ERROR_DEFER_DRIVER_INIT if interrupt controller is not yet inited
+ * Return TEE_ERROR_ITEM_NOT_FOUND if the DT does not reference target interrupt
+ * Return any other TEE_Result compliant code in case of error
+ */
+TEE_Result dt_get_interrupt_by_index(const void *fdt, int node,
+				     unsigned int index, struct itr_chip **chip,
+				     size_t *itr_num);
+
+/**
+ * dt_get_interrupt_by_name() - Get an interrupt from DT by interrupt name
+ *
+ * @fdt		Device tree to work on
+ * @node	Node offset of the subnode containing interrupt(s) references
+ * @name	Name identifier used in "interrupt-names" property
+ * @chip	Output interrupt controller clock reference upon success
+ * @itr_num	Output interrupt line upon success
+ *
+ * Return TEE_SUCCESS in case of success
+ * Return TEE_ERROR_DEFER_DRIVER_INIT if interrupt controller is not yet inited
+ * Return TEE_ERROR_ITEM_NOT_FOUND if the DT does not reference target interrupt
+ * Return any other TEE_Result compliant code in case of error
+ */
+TEE_Result dt_get_interrupt_by_name(const void *fdt, int node, const char *name,
+				    struct itr_chip **chip, size_t *itr_num);
+#else
+static inline
+TEE_Result dt_register_interrupt_provider(const void *dt __unused,
+					  int node __unused,
+					  dt_get_itr_func dt_get_itr __unused,
+					  void *data __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+static inline
+TEE_Result dt_get_interrupt_by_index(const void *fdt __unused,
+				     int node __unused,
+				     unsigned int index __unused,
+				     struct itr_chip **chip __unused,
+				     size_t *itr_num __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+static inline TEE_Result dt_get_interrupt_by_name(const void *fdt __unused,
+						  int node __unused,
+						  const char *name __unused,
+						  struct itr_chip **ch __unused,
+						  size_t *itr_num __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+#endif /*CFG_DT*/
+
+/* Helper function for when interrupt index is 0 */
```

Comment:
What's an interrupt index? Is it something DT specific?

Reply:
Interrupt index relates to the index of the interrupt reference by DT binding property `interrupts` that can list several interrupts a driver can consume , see [DT binding docs](https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/interrupt-controller/interrupts.txt#L17). It is not related to interrupt number identifier value used by an interrupt controllers.

Reply:
Thanks for the explanation. Please add something short about what index means.